packages/core/src/render3/instructions/shared.ts

```diff
@@ -1692,7 +1707,16 @@ export function setElementAttribute(
     const strValue =
       sanitizer == null ? renderStringify(value) : sanitizer(value, tagName || '', name);
 
-    renderer.setAttribute(element, name, strValue as string, namespace);
+    setAttributeWithHydrationSupport(
```

Comment:
I'd propose renaming this method to something like this:
```suggestion
    _setAttributeImpl(
```
This should make it more clear that the implementation is tree-shakable and hydration code paths are not invoked for client-only apps.

Reply:
Updated.